src/test/java/teammates/test/driver/RetryManager.java

```diff
@@ -0,0 +1,100 @@
+package teammates.test.driver;
+
+import teammates.common.util.Assumption;
+import teammates.common.util.ThreadHelper;
+
+/**
+ * Handles running and retrying of {@code Retryable} tasks.
+ */
+public final class RetryManager {
```

Comment:
Can we put RetryManager and its class to a new package? (To make it well-organized, also we could enforce architectural rule here)